### ISSUE TITLE
[Modded] Searching for items in fabricators prevents interaction (2 FPS lag), can't exit fabricator UI

### ISSUE THREAD

**Issue by Krzeszny** [Regalis11/Barotrauma#9587]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?
Edited:
When you type anything in the search bar, the game gradually starts slowing down to 2 FPS and ignores mouse and keyboard input until you're stunned or dragged away. 
When you're eventually dragged away and reopen the GUI, the lag prevents you even from clearing the search bar, so fabricators become unusable. No workaround found. It happens every round in which you use the search bar.

Possible causes:
a) there are too many items from mods listed in a fabricator, including **Enhanced Armaments** (most likely)
b) there are too many items in a linked cabinet (less likely)
c) multiplayer (maybe exclusively)
d) too many items on the ship

### Reproduction steps
Edited:
Not sure. Doesn't happen in the editor and not on all servers, and not to all players. Happens with both fabricators. I've only seen this problem with the mod Enhanced Armaments.

### Bug prevalence

Happens regularly in certain modded campaigns

### Version

Other

### -

0.18.15.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

_Stand by_

**Comment by Ek-Videogames** [comment]
Client reproduction without any linked containers - just fab UI, think the slowdown could be recipe updates?
Maybe result of the increase in recipe count from recycle-recipes.

**Comment by Regalis11** [comment]
Hmm, it seems selecting an item from the list does cause a noticeable performance drop due to the logic that highlights available ingredients in the linked containers. I couldn't get it anywhere near 2 FPS though, just dropped from ~140 FPS to ~120 FPS.

**Comment by Regalis11** [comment]
I implemented some optimizations to the highlighting logic in https://github.com/Regalis11/Barotrauma-development/commit/b428f8e3aa38c3367ad1b8f0b3692be358f48856, let's see if this helps when we get it out to unstable

**Comment by Krzeszny** [comment]
> So it seems some other conditions are required to get this to happen. Would it be possible to send us one of these campaign saves where this occurs? A screenshot with showperf enabled might also give us some more clues. Did it happen in single player or multiplayer?

- I don't play singleplayer. It happens in co-op and not to all players, it seems.
- So far it's only been happening with the mod Enhanced Armaments installed.
- I wasn't able to reproduce it in singleplayer either.
- I don't have any showperf screenshots or videos but I remember I was observing showperf and there weren't any noticeable increases

**Comment by Krzeszny** [comment]
https://user-images.githubusercontent.com/10529134/180866154-64a6b33a-cc63-45d7-9c5e-ae4deeb37b32.mp4

An example from today. It's been fine for several minutes and suddenly the UI/mouse cursor FPS decreased to 1 per several seconds, while in-game FPS decreased to about 8.

**Comment by Rokvach** [comment]
After the fix, only time I noticed a noticeable dip in performance was when searching the fabricator for the very first time after launching the game and having a lot of mods enabled which added items to the fabricator list. It was just a frame drop which lasted maybe a second and then returned to normal. There weren't any framerate issues after that first time and the fps stayed constant no matter what I searched.
